static/js/pv_tree.tsx

```diff
@@ -0,0 +1,163 @@
+import React, {Component} from "react";
+import hierarchy from "../../tools/pv_tree_generator/hierarchy_golden.json";
+//import hierarchy from "../data/hierarchy.json";
+
+// Fields of each node from json file
+interface NodeType {
+  show: string; 
+  title: string; 
+  selected: string;
+  expanded: string;
+  children: NodeType[];
+  count: number;
+  search_count: number;
+  measuredProperty: string;
+  placeTypes: string[];
+  populationType: string;
+  type: string; // Property or Value
+  argString: string;// for statvars, argString = dcid
+}
+
+interface NodePropType {
+  title: string; 
+  selected: string;
+  expanded: string;
+  count: number;
+  children: NodeType[];
+  type: string;
+  argString: string;
+  updateurl: (string, boolean) => void;
+  nodePath: string;
+}
+
+class Node extends Component<NodePropType, {}> {
+  constructor(props) {
+    super(props);
+    this._handleCheckboxClick = this._handleCheckboxClick.bind(this);
+    this._handleExpandClick = this._handleExpandClick.bind(this);
+    this.state = {
+      checked: props.selected === "yes" ? true: false,
+      expanded: props.selected === "yes" ? true: false,
+      nodePath: props.nodePath + "," + props.title,
+    };
+  }
+
+  private _handleCheckboxClick = ():void => {
+    this.setState({
+        checked: !this.state.checked,
+    });
+    this.props.updateurl(this.props.argString +this.state.nodePath, !this.state.checked);
+  }
+
+  private _handleExpandClick = ():void => {
+    this.setState({
+        expanded: !this.state.expanded,
+      });
+  }
+
+  render() {
+    let checkboxImg;
+    let expandImg;
+    let child; 
+    if (this.props.type === "value") {
+      checkboxImg = <button className={this.state.checked? 
+                                        "checkbox checked":"checkbox"}  
+                              onClick={this._handleCheckboxClick}/>};
+
+    if (this.state.expanded) {
```

Comment:
this is quite interesting, so you only create node when it is expanded, then the actual created nodes should be much less than before where we build the entire tree?

Reply:
right.